src/constants/TokenRestrictionLookupTable.ts

```diff
@@ -0,0 +1,32 @@
+import { SupportedChainId } from './chains'
+import { TOKEN_RESTRICTION_TYPE, TOKEN_RESTRICTIONS } from './tokenRestrictions'
+
+class TokenRestrictionLookupTable {
+  dict: { [chainId: number]: { [key: string]: TOKEN_RESTRICTION_TYPE } | null } | null = null
+
+  createMap() {
+    const dict: { [chainId: number]: { [key: string]: TOKEN_RESTRICTION_TYPE } | null } | null = {}
+
+    for (const chain of Object.keys(TOKEN_RESTRICTIONS).map((t) => parseInt(t))) {
+      const tokens = Object.keys(TOKEN_RESTRICTIONS[chain as SupportedChainId] ?? {})
+      for (const token of tokens) {
+        if (!dict[chain]) dict[chain] = {}
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        dict[chain]![token.toLowerCase()] = TOKEN_RESTRICTIONS[chain as SupportedChainId]![token]
```

Comment:
I think this one is a difficult one. The null checks already happen on line 13, but since it's a mapping, the access can only work inline AFAIK and assigning variables copies by value instead.

Open to suggestions to making this better though if you know a way

Reply:
could we make it something like:
`if (!dict[chain]) { 
dict[chain] = {} 
} else if (dict[chain][token.toLowerCase()] { 
    if(TOKEN_RESTRICTIONS[chain as SupportedChainId] { 
        dict[chain][token.toLowerCase()] = TOKEN_RESTRICTIONS[chain as SupportedChainId]![token] 
        }}`

Seems like line breaks are not working correctly here, sorry about it

Reply:
All good :) You can use triple backticks for codeblocks with line breaks and indents :D

So should be:

```
if (!dict[chain]) {  
    dict[chain] = {}
} else if (dict[chain][token.toLowerCase()]) {
    if(TOKEN_RESTRICTIONS[chain as SupportedChainId] {
        dict[chain][token.toLowerCase()] = TOKEN_RESTRICTIONS[chain as SupportedChainId]![token]
    }
}
```

Unfortunately it looks like we can't get rid of non-null assertions for this `dict[chain]` so we'll always need it to be `dict[chain]!`. We also don't want to be checking if `dict[chain][token.toLowerCase()]` exists because either way we are setting this value.

Otherwise to me it actually looks the same (ends up having the same line of code but like wrapped in more if statements which makes it longer IMO)